platform/ethereum/compound/compound.go

```diff
@@ -0,0 +1,216 @@
+package compound
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+)
+
+// Compound Lending Provider
+
+type Provider struct {
+	client Client
+}
+
+func Init(api string) *Provider {
+	return &Provider{
+		client: Client{blockatlas.InitClient(api)},
+	}
+}
+
+const (
+	_providerName                   = "compound"
+	_cacheValiditySec time.Duration = 15 * time.Second
+)
+
+func (p *Provider) Name() string {
+	return _providerName
+}
+
+// GetProviderInfo return static info about the lending provider, such as name and asset classes supported.
+func (p *Provider) GetProviderInfo() (blockatlas.LendingProvider, error) {
+	prov := blockatlas.LendingProvider{
+		ID: "compound",
+		Info: blockatlas.LendingProviderInfo{
+			ID:          _providerName,
+			Description: "Compound Decentralized Finance Protocol",
+			Image:       "https://compound.finance/images/compound-logo.svg",
+			Website:     "https://compound.finance",
+		},
+		Type:   blockatlas.ProviderTypeLending,
+		Assets: []blockatlas.AssetInfo{},
+	}
+	assets, err := p.getAssetInfos(false)
+	if err != nil {
+		prov.Assets = assets
+	}
+	return prov, nil
+}
+
+// GetAsset return asset info including APY.  Rates are annualized.  Rates vary over time.
+// asset: Symbol to consider, or token address.  Can be empty, then all are returned.
+func (p *Provider) GetAsset(asset string) ([]blockatlas.AssetInfo, error) {
```

Comment:
let's make this function more intuitive to use and only return specific asset or all assets, otherwise it's confusing that empty will return all

Reply:
done